### PR TITLE
Fixes scale of input data

### LINKED ISSUE #12: `Error: $ operator is invalid for atomic vectors`
The data in the branch `7-restructure-package` throws an error 

```
> out_bayes1 <- BayesLogit::logit(y=0.5*(dataset1$y_train+1), X=cbind(1,dataset1$x_train), P0=diag(rep(1/prior_variance,dataset1$n_cov+1) ), samp=n_s .... [TRUNCATED] 
[1] "y is a proportion; it must be <= 1."
Error: $ operator is invalid for atomic vectors
> traceback()
5: BayesLogit::logit(y = y, X = X, P0 = P0, samp = n_samp, burn = n_samp) at pcb_vb_classn_sb.R#224
4: eval(ei, envir)
3: eval(ei, envir)
2: withVisible(eval(ei, envir))
1: source("~/Projects/PosteriorBootstrap/PosteriorBootstrap/R/pcb_vb_classn_sb.R", 
       echo = TRUE)
```

Scaling the `y` argument to be between 0 and 1 fixes the issue. I don't know if we can debug inside the `BayesLogit` dataset, otherwise we need to ask Simon for his original data.

### PR DESCRIPTION
**Pull request by mm3509**
Closes #12 .

# tl;dr

The outcome in the original data is {1, 2} (1 for good credit, 2 for bad credit).

Line 45 was converting it to {1, 3}. Line 212 was converting {1, 3} to {1, 2}, which is not a probability.

The comment on line 45 said "y is now in +/- 1 format". Fixing that transformation to really be to {-1, +1} fixed the problem, as line 212 now maps to {0, 1}.

In addition, I added the library e1071 and changed the paths to save the figure and the workspace. The code takes about 3 hours to run and the result is similar to Figure 2 in the paper, although not exactly equal despite setting the seeds.

# full version

## Simon Lyddon's paper

> We demonstrate this in practice through a VB logistic regression model fit to the Statlog German Credit dataset, containing 1000 observations and 25 covariates (including intercept), from the UCI ML repository [12], preprocessing via [13].

and the references are:

>[12] Dua Dheeru and Efi Karra Taniskidou. UCI Machine Learning Repository, 2017.

> [13] Manuel Fernández-Delgado, Eva Cernadas, Senén Barro, and Dinani Amorim. Do
we Need Hundreds of Classifiers to Solve Real World Classification Problems?
Journal of Machine Learning Research, 15:3133–3181, 2014.

## UCI Machine Learning Repository

The [UCI Machine Learning Repository](http://archive.ics.uci.edu/ml/datasets.html) has these details for "Statlog (German Credit Data)":
	
Multivariate 
	
Classification 
	
Categorical, Integer 
	
1000 instances
	
20 covariates
	
1994

## Dataset page

The [dataset page](http://archive.ics.uci.edu/ml/datasets/Statlog+%28German+Credit+Data%29) shows:

> This dataset requires use of a cost matrix (see below)

```
..... 1 2
----------------------------
1 0 1
-----------------------
2 5 0

(1 = Good, 2 = Bad) 
```

> The rows represent the actual classification and the columns the predicted classification.

> It is worse to class a customer as good when they are bad (5), than it is to class a customer as bad when they are good (1).

That is, a code of 1 means "good credit", a code of 2 means "bad credit". Correct classification incurs 0 cost, classifying good as bad incurs a cost of 5, and classifying bad as good incurs a cost of 1.

Although the page does not say so, the labels of good and bad are in the very last column,  after the label on foreign or domestic worker. If you look at the [categorical data](http://archive.ics.uci.edu/ml/machine-learning-databases/statlog/german/german.data), you will see a second to last column with `A201` or `A202` for foreigner or not, and the last column with only 1's and 2's. So I assume that this is the outcome column because this is the only column that does not contain a description in the dataset page. Furthermore, only columns 18 and 21 are between 1 and 2, and attribute 18 is described in the dataset as sitting between attribute 17 for "Job" and attribute 18 for  "Telephone", and the categorical values in the data in column 17 correspond to attribute 17 and same for those in column 19.

## Pre-processing

Simon's paper quotes [Manuel Fernández-Delgado, Eva Cernadas, Senén Barro, and Dinani Amorim. Do we Need Hundreds of Classifiers to Solve Real World Classification Problems?](http://jmlr.csail.mit.edu/papers/v15/delgado14a.html), whose only pre-processing part is:

> Each input is pre-processed to have zero mean and standard deviation one, as is usual in the classifier literature. We do not use further pre-processing, data transformation or feature selection. 

But this is not the pre-processing done in the code: the transformed data has mean 0.3 and variance 0.2.

See the above `tl;dr` for the solution

### CODE REVIEW

**Comment by slllll** [comment]
It doesn't seem like you're using the same starting dataset. See my comments on #15

**Comment by miguelmorin** [comment]
Indeed, we were not. See my comments in #12, with the PR #31 that fixes this.
